crates/bitwarden-ipc/src/traits/communication_backend.rs

```diff
@@ -20,7 +20,27 @@ pub trait CommunicationBackend {
     fn receive(
         &self,
     ) -> impl std::future::Future<Output = Result<IncomingMessage, Self::ReceiveError>>;
+
+    /// Subscribe to receive messages. This function will return a receiver that can be used to
+    /// receive messages asynchronously. Multiple receivers may be created, and all receivers will
+    /// receive the same messages.
+    ///
+    /// Use this function if you are expecting to receive a message as a response to a message you
+    /// sent. Subscribing will start buffering messages and help you avoid missing messages.
+    fn subscribe(&self) -> impl CommunicationBackendReceiver<ReceiveError = Self::ReceiveError>;
```

Comment:
As far as I understand it, the invariant that all the subscribers will receive all the same messages is something that the trait implementations need to enforce right? If so it might be a good idea to mention it in here. Otherwise someome might think that it's the `IpcClient` that handles that for you, leading to inconsistent implementations.

The same also probably applies to `CommunicationBackendReceiver::receive`

Reply:
Good point, I've updated the doc, thoughts?